README.md

```diff
@@ -25,464 +24,431 @@ Features include:
-
-- Friendly APIs with sane defaults
-- Extensible without breaking clients or polluting the global namespace with too many constructors
-- Options are immutable, reusable, and composable
-
-They are easy to use and look like this:
+The Huma router is the entrypoint to your service or application. There are a couple of ways to create it, depending on what level of customization you need.
 
 ```go
-// Add a parameter with an example
-huma.PathParam("id", "Resource identifier", huma.Example("abc123"))
-```
+// Simplest way to get started, which creats a router and a CLI with default
+// middleware attached. Note that the CLI is a router.
+app := cli.NewRouter("API Name", "1.0.0")
 
-Most text editors will auto-complete and show only the available options, which is an improvement over e.g. accepting `interface{}`.
+// Doing the same as above by hand:
+router := huma.New("API Name", "1.0.0")
+app := cli.New(router)
+middleware.Defaults(app)
 
-### Extending & Composition
-
-Functional options can be wrapped to extend the set of available options. For example:
-
-```go
-// IDParam creates a new path parameter limited to characters and a length that
-// is allowed for resource identifiers.
-func IDParam(name, description string) huma.DependencyOption {
-	s := schema.Schema{Pattern: "^[a-zA-Z0-9_-]{3,20}"}
-
-	return huma.PathParam(name, description, huma.Schema(s))
-}
+// Start the CLI after adding routes:
+app.Run()
 ```
 
-You can also compose multiple options into one, e.g by using `huma.ResourceOptions(..)` or one of the other related functions:
+You can also skip using the built-in `cli` package:
 
 ```go
-// CommonOptions sets up common options for every operation.
-func CommonOptions() huma.ResourceOption {
-	return huma.ResourceOptions(
-		huma.Tags("some-tag"),
-		huma.HeaderParam("customer", "Customer name", "", huma.Internal()),
-		huma.ResponseError(http.StatusInternalServerError, "Server error"),
-	)
-}
+// Create and start a new router by hand:
+router := huma.New("API Name", "1.0.0")
+router.Middleware(middleware.DefaultChain)
+router.Listen("127.0.0.1:8888")
 ```
 
 ## Resources
 
-Huma APIs are composed of resources and sub-resources attached to a router. A resource refers to a unique URI on which operations can be performed. Huma resources can have dependencies, security requirements, parameters, response headers, and responses attached to them which are all applied to every operation and sub-resource.
+Huma APIs are composed of resources and sub-resources attached to a router. A resource refers to a unique URI on which operations can be performed. Huma resources can have middleware attached to them, which run before operation handlers.
```

Comment:
> which run before operation handlers.
Is my understanding correct that the middleware executes to completion, then the operation handler is executed? 

No change requested, just looking to understand: I am mentally contrasting it with a gRPC interceptor where the middleware _calls_ the handler so it can also take action on the handler response.

Reply:
It's the same concept. Each middleware takes a `next` handler and calls `next.ServeHTTP(w, r)` on it, and at the end of the chain `next` is just the handler function itself.